matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/util/TemporaryFileCreator.kt

```diff
@@ -28,7 +28,10 @@ internal class TemporaryFileCreator @Inject constructor(
 ) {
     suspend fun create(): File {
         return withContext(Dispatchers.IO) {
-            File.createTempFile(UUID.randomUUID().toString(), null, context.cacheDir)
+            runCatching {
+                File.createTempFile(UUID.randomUUID().toString(), null, context.cacheDir)
+                        .apply { mkdirs() }
+            }.getOrThrow()
```

Comment:
runCatching / getOrThrow thing :)

Reply:
In this case I'm not pro for making it `getOrNull`. Trying it out, it leads to a long chain of nullable changes which could lead to some weird bugs that are harder for us to identify than a crash.

I doubt that we should be crashing here though. given that it's effectively the same as `File.createTempFile` on its own. Just without the warning shown below
![image](https://user-images.githubusercontent.com/20701752/163357670-aad788e8-e470-425a-bc17-b8b94376ad78.png)

Reply:
Using `runCatching / getOrThrow` just to get ride of the warning is quite strange to me. We do other people think?